MoEngagePluginCards/MoEngagePluginCardsBridge.swift

```diff
@@ -0,0 +1,443 @@
+                attribute: accountData
+            )
+        else {
+            logAppIdentifierFetchFailed(for: accountData)
+            return
+        }
+
+        MoEngageSDKCards.sharedInstance.onAppOpenSync(forAppID: identifier) { data in
+            self.syncManager.sendUpdate(
+                forEventType: .inboxOpen,
+                andAppID: identifier,
+                withNewData: data
+            )
+        }
+    }
+
+    @objc public func onCardsSectionUnLoaded(_ accountData: [String: Any]) {
+        guard
+            let identifier = MoEngagePluginUtils.fetchIdentifierFromPayload(
+                attribute: accountData
+            )
+        else {
+            logAppIdentifierFetchFailed(for: accountData)
+            return
+        }
+        self.handler.cardsViewControllerDismissed(forAppID: identifier)
+    }
+
+    @objc public func cardClicked(_ cardData: [String: Any]) {
+        guard
+            let identifier = MoEngagePluginUtils.fetchIdentifierFromPayload(
+                attribute: cardData
+            )
+        else {
+            logAppIdentifierFetchFailed(for: cardData)
+            return
+        }
+
+        let cardClick: MoEngageCardClickData
+        do {
+            let clickData: [String: Any] = try MoEngagePluginCardsUtil.getData(fromHybridPayload: cardData)
+            cardClick = try MoEngageCardClickData.decodeFromHybrid(clickData)
+        } catch {
+            MoEngageLogger.error("\(error)")
+            return
+        }
+
+        if let widgetId = cardClick.widgetId {
+            self.handler.cardClicked(
+                cardClick.card, withWidgetID: widgetId,
+                forAppID: identifier
+            )
+        } else {
+            self.handler.cardClicked(cardClick.card, forAppID: identifier)
+        }
+    }
+
+    @objc public func cardDelivered(_ accountData: [String: Any]) {
+        guard
+            let identifier = MoEngagePluginUtils.fetchIdentifierFromPayload(
+                attribute: accountData
+            )
+        else {
+            logAppIdentifierFetchFailed(for: accountData)
+            return
+        }
+        self.handler.cardDelivered(forAppID: identifier)
+    }
+
+    @objc public func cardShown(_ cardData: [String: Any]) {
+        guard
+            let identifier = MoEngagePluginUtils.fetchIdentifierFromPayload(
+                attribute: cardData
+            )
+        else {
+            logAppIdentifierFetchFailed(for: cardData)
+            return
+        }
+
+        let card: MoEngageCardCampaign
+        do {
+            let showData: [String: Any] = try MoEngagePluginCardsUtil.getData(fromHybridPayload: cardData)
+            card = try MoEngageHybridSDKCards.buildCardCampaign(fromHybridData: showData)
+        } catch {
+            MoEngageLogger.error("\(error)")
+            return
+        }
+        self.handler.cardShown(card, forAppID: identifier)
```

Comment:
@msoumya-engg-sdk can u check in all places, where u can move line 186 inside do block and avoid extra variable declaration

Reply:
@RakshithaAcharya made changes